File: app/src/main/java/com/datarockets/mnchkn/views/fonts/FontCache.java
package com.datarockets.mnchkn.views.fonts;

import android.content.Context;
import android.graphics.Typeface;

import java.util.HashMap;

import timber.log.Timber;

public class FontCache {

    private static HashMap<String, Typeface> fontNameTypefaceMap = new HashMap<>();

    public static Typeface getTypeface(String fontName, Context context) {
        Typeface typeface = fontNameTypefaceMap.get(fontName);
        if (typeface == null) {
            try {
                typeface = Typeface.createFromAsset(context.getAssets(), fontName);
            } catch (Exception e) {
                Timber.e("Error while trying to get typeface");
            }
            fontNameTypefaceMap.put(fontName, typeface);
        }
        return typeface;
    }

}

File: app/src/main/java/com/datarockets/mnchkn/views/fonts/FontCache.kt
package com.datarockets.mnchkn.views.fonts

import android.content.Context
import android.graphics.Typeface
import timber.log.Timber
import java.util.*

object FontCache {

    private val fontNameTypefaceMap = HashMap<String, Typeface>()

    fun getTypeface(fontName: String, context: Context): Typeface {
        var typeface: Typeface? = fontNameTypefaceMap[fontName]
        if (typeface == null) {
            try {
                typeface = Typeface.createFromAsset(context.assets, fontName)
            } catch (e: Exception) {
                Timber.e("Error while trying to get typeface")
            }
            fontNameTypefaceMap.put(fontName, typeface!!)
        }
        return typeface
    }
}
